File: Fase 2/Evidencias Proyecto/Evidencias de sistema/InventaPro/src/app/interfaces/perdida.ts
import { Bodega } from "./bodega";
import { Producto } from "./producto";

export interface Perdida{
  perdidaId: number;
  productoId: number;
  producto: Producto;

  bodegaId: number;
  bodega: Bodega;

  cantidad: number;
  tipoPerdidaId: number;
  tipoPerdida: TipoPerdida;

  descripcion: string;
  fechaRegistro: string;
  usuarioId: number;
  precioCompra: number;
}

export interface TipoPerdida {
    tipoPerdidaId: number;
    nombre: string;
}

export interface PerdidaCreate{
  perdidas: {
    productoId: number;
    bodegaId: number;
    cantidad: number;
    tipoPerdida: number;
    descripcion: string;
    precioCompra: number;
  }[];
}

File: Fase 3/Evidencias Grupales/Evidencias Proyecto/Evidencias de sistema/InventaProAPI/Models/Perdida.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace InventaProAPI.Models
{
  [Table("Perdidas")]
  public class Perdida
  {
    [Key]
    [Column("perdida_id")]
    public int PerdidaId { get; set; }

    [Required]
    [Column("producto_id")]
    [ForeignKey("Producto")]
    public int ProductoId { get; set; }

    [Required]
    [Column("bodega_id")]
    [ForeignKey("Bodega")]
    public int BodegaId { get; set; }

    [Required]
    [Column("cantidad")]
    public int Cantidad { get; set; }

    [Required]
    [Column("tipo_perdida_id")]
    [ForeignKey("TipoPerdida")]
    public int TipoPerdidaId { get; set; }

    [Column("descripcion")]
    public string Descripcion { get; set; }

    [Column("fecha_registro")]
    public DateTime FechaRegistro { get; set; }

    [Required]
    [Column("usuario_id")]
    [ForeignKey("Usuario")]
    public int UsuarioId { get; set; }

    [Required]
    [Column("precio_compra")]
    public int PrecioCompra {  get; set; }

    public Producto Producto { get; set; }

    public Bodega Bodega { get; set; }

    public TipoPerdida TipoPerdida { get; set; }

    [JsonIgnore]
    public Usuario Usuario { get; set; }
  }
}
